engine/src/main/java/org/hibernate/search/analyzer/impl/RemoteAnalyzerProvider.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.analyzer.impl;
+
+/**
+ * Applies to index managers that are bound to backends supporting built-in analyzer definitions.
+ * <p>
+ * For example, Elasticsearch comes with whitespace and languages analyzers without a definition in the classpath.
+ *
+ * @author Davide D'Alto
+ */
+public interface RemoteAnalyzerProvider {
```

Comment:
How about adding a method

    AnalyzerReference getRemoteAnalyzer(String name);

For now this would just return a new instance without any validation, but as of ES 2.3 (or whenever that bug is fixed), it can be enriched to actually check the given name is valid and raise an exception otherwise.